### PR TITLE
#3433 Search bar suggestions changes

### LINKED ISSUE #3433: Search bar suggestions in explore/collections
we have search bar on collections hotfixed thanks to:
- https://github.com/kodadot/nft-gallery/pull/3390

next thing would be to get rid of the errors, after speaking to @vikiival, the problem is with magick (subquery) queries being passed to our [snek indexer](https://github.com/kodadot/snek), and if I understand the problem correctly, moving forward we should rewrite the queries to subsquid so that we don't get those graphql errors we now get when typing + there seems to be some redundant redirection happening:
image

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #3433
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-07-19 at 7 36 31 PM

### CODE REVIEW

**Review by cryptodamsky — APPROVED** [review]
Looks good

**Review by petersopko — CHANGES_REQUESTED** [review]
when I'm clicking on suggested item from landing page, I'm weirdly redirected to search of the term I've typed in, not to the actual item.
Also this shows up:
image

**Comment by prachi00** [comment]
> when I'm clicking on suggested item from landing page, I'm weirdly redirected to search of the term I've typed in, not to the actual item. Also this shows up: image

@petersopko try now. should work

**Comment by petersopko** [comment]
> @petersopko try now. should work

ok, the routing and suggestions now works as expected, last bit following up in here:

- https://github.com/kodadot/nft-gallery/issues/3504

**Comment by petersopko** [comment]
let's wait until @Jarsen136 checks the code and merge if there's nothing left to fix :)

File: components/rmrk/Gallery/Search/SearchBar.vue
Comment (Jarsen136) [inline components/rmrk/Gallery/Search/SearchBar.vue:502]: I am not sure why moving this function `this.redirectToGalleryPageIfNeed()`

Comment (prachi00) [reply components/rmrk/Gallery/Search/SearchBar.vue:502]: Because it gets called everytime after you focus and then click anywhere outside, it should only be called in case of enter pressed

**Comment by prachi00** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 61.49 USD/KSM ~ 1.626 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0xf32b1ee94897e14378fcc86de4b9d916c982a13c2c9d922e30fc9e6be306830b](https://kusama.subscan.io/extrinsic/0xf32b1ee94897e14378fcc86de4b9d916c982a13c2c9d922e30fc9e6be306830b)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**
